common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleIndexCache.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.shuffle;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.LongBuffer;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Maintains an LRU cache of {@link ShuffleIndexInformation} so that
+ * we can avoid open/close of the index files for each block fetch.
+ */
+public class ShuffleIndexCache {
+
+  LoadingCache<String, ShuffleIndexInformation> indexCache;
+
+  public ShuffleIndexCache(long cacheSize) {
+    CacheLoader<String, ShuffleIndexInformation> loader =
```

Comment:
nit: does it work to have `File` as a key? That seems cleaner than a raw string.